src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,394 @@
+     * @throws IOException launched if a problem exists reading the PEM information or the {@link File}
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull File pemFile, @Nullable String passphrase) throws IOException {
+        decodePEM(pemFile, passphrase);
+    }
+
+    /**
+     * Creates a {@link PEMManager} by reading PEM formated data from a {@link String}
+     * 
+     * @param pem {@link String} with the PEM data
+     * @throws IOException launched if a problem exists reading the PEM information
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull String pem) throws IOException {
+        decodePEM(pem, null);
+    }
+
+    /**
+     * Creates a {@link PEMManager} by reading PEM formated data from a {@link String}
+     * 
+     * @param pem {@link String} with the PEM data
+     * @param passphrase passphrase for the encrypted PEM data. null if PEM data is not passphrase protected
+     * @throws IOException launched if a problem exists reading the PEM information
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull String pem, @Nullable String passphrase) throws IOException {
+        decodePEM(pem, passphrase);
+    }
+
+    /**
+     *
+     * Creates a {@link PEMManager} from an {@link Object} that can be of any supported type by Bouncy Castle or JCA:
+     * {@link Key}, {@link PrivateKey}, {@link KeyPair}, {@link Certificate}, etc.
+     *
+     * @param pemObject object to manage
+     */
+    public PEMManager(@Nonnull Object pemObject) {
+        this.pemObject = pemObject;
+    }
+
+    private void decodePEM(@Nonnull File pemFile, @Nullable String passphrase) throws IOException {
+        decodePEM(FileUtils.readFileToString(pemFile), passphrase);
+    }
+
+    private void decodePEM(@Nonnull String pem, @Nullable final String passphrase) throws IOException {
+        PasswordFinder pwf = null;
+        if (passphrase != null) {
+            pwf = new PasswordFinder() {
+                @Override
+                public char[] getPassword() {
+                    return passphrase.toCharArray();
+                }
+            };
+        }
+
+        PEMReader parser = new PEMReader(new StringReader(pem), pwf);
+        try {
+            pemObject = parser.readObject();
+        } catch (PasswordException pwE) {
+            throw new BCPasswordException(pwE);
+        } finally {
+            parser.close();
+        }
+    }
+
+    /**
+     * Encodes the current stored information in PEM format and returns it as a {@link String}
+     * 
+     * @return PEM encoded data
+     * @throws IOException launched if a problem exists generating the PEM information
+     */
+    @Nonnull
+    public String encodePEM() throws IOException {
```

Comment:
the API feels unbalanced between these methods and the constructors. Perhaps just have a single constructor that takes the object and then make the `decode` methods static and returning the instance.

Also there is redundant use of PEM in the name... it's going to get really tiring typing

`PEMEncodableObject.decodePEM(string)`

Better would be `PEMEncodableObject.decode(string)`

Then we say... well `Object` is redundant so a better name for the class is `PEMEncodable`